e2e-tests/test/hardhat-apis.test.ts

```diff
@@ -84,3 +88,34 @@ xdescribe("hardhat_impersonateAccount & hardhat_stopImpersonatingAccount", funct
     expect(await provider.getBalance(RichAccounts[0].Account)).to.equal(beforeBalance.sub(0.42));
   });
 });
+
+describe("hardhat_setCode", function () {
+  it("Should update code with a different smart contract", async function () {
```

Comment:
Can we add a test for passing in an address that isn't assigned to a contract/account?